packages/teams-js/src/internal/communication.ts

```diff
@@ -712,6 +744,18 @@ function handleParentMessage(evt: DOMMessageEvent): void {
       logger('Removing registered promise callback for message %i', message.id);
       delete CommunicationPrivate.promiseCallbacks[message.id];
     }
+    const portCallback = CommunicationPrivate.portCallbacks[message.id];
+    if (portCallback) {
+      logger('Invoking the registered port callback for message %i with arguments %o', message.id, message.args);
+      let port: MessagePort | null = null;
+      if (evt.ports && evt.ports[0] instanceof MessagePort) {
+        port = evt.ports[0];
```

Comment:
It would be helpful to throw an exception here if no port is passed down, rather than where you are doing it in `messageChannel.ts`. I'm don't anticipate a lot of functions will be requesting ports from the host, but just in case it would be good to check for that common error and throw here rather than depend on all of the callers to handle it consistently.

Reply:
Makes sense.

Reply:
Realized that throwing it here wouldn't actually trickle back to the caller, have put it in the callback itself.